src/Cryptol/IR/FreeVars.hs

```diff
@@ -52,9 +53,9 @@ rmVals p x = x { valDeps = Set.difference (valDeps x) p }
 -- | Compute the transitive closure of the given dependencies.
 transDeps :: Map Name Deps -> Map Name Deps
 transDeps mp0 = fst
-              $ head
+              $ headInfList
               $ dropWhile (uncurry (/=))
-              $ zip steps (tail steps)
+              $ zipTail steps
```

Comment:
Could this just be `zip steps $ drop 1 steps` to avoid having to define `zipTail`?

Reply:
Yes, I like your version better.